bpf/tests/pktgen.h

```diff
@@ -938,6 +938,98 @@ static __always_inline void pktgen__finish_ipv6_opt(const struct pktgen *builder
 	}
 }
 
+static __always_inline __u32
+pktgen__ip_csum(const struct pktgen *builder, int i)
+{
+	__u32 csum;
+	__u32 tmp;
+
+	switch (builder->layers[i - 1]) {
+	case PKT_LAYER_IPV4:
+		if (builder->layer_offsets[i - 1] >= MAX_PACKET_OFF - sizeof(struct iphdr))
+			return 0;
+
+		struct iphdr *ipv4_layer;
+
+		ipv4_layer = ctx_data(builder->ctx) + builder->layer_offsets[i - 1];
+		if ((void *)ipv4_layer + sizeof(struct iphdr) > ctx_data_end(builder->ctx))
```

Comment:
just a question, don't we already have some common functions to do the header validation - or is this just to keep this code distinct from general bpf lib code?

Reply:
@tommyp1ckles I don't think I get the question.Could you please elaborate on it?